app/models/manageiq/providers/vmware/manager_auth_mixin.rb

```diff
@@ -36,10 +44,23 @@ def connect(options = {})
     self.class.raw_connect(server, port, username, password)
   end
 
+  def translate_exception(err)
+    case err
+      when Fog::Compute::VcloudDirector::Unauthorized
+        MiqException::MiqInvalidCredentialsError.new "Login failed due to a bad username or password."
+      when Excon::Errors::Timeout
+        MiqException::MiqUnreachableError.new "Login attempt timed out"
+      when Excon::Errors::SocketError
+        MiqException::MiqHostError.new "Socket error: #{err.message}"
+      when MiqException::MiqInvalidCredentialsError, MiqException::MiqHostError # from amqp
```

Comment:
Are these really only possible from the AMQP? The error names indicate they may be more generic, so in this case I would suggest to remove the `from amqp` comment.